src/libs/ReportUtils.ts

```diff
@@ -23,171 +26,193 @@ import isReportMessageAttachment from './isReportMessageAttachment';
+    name: string;
+    fallbackIcon?: React.FC<SvgProps> | string;
+};
+
+type ExpanseOriginalMessage = {
+    oldComment?: string;
+    newComment?: string;
+    comment?: string;
+    merchant?: string;
+    oldCreated?: string;
+    created?: string;
+    oldMerchant?: string;
+    oldAmount?: number;
+    amount?: number;
+    oldCurrency?: string;
+    currency?: string;
+    category?: string;
+    oldCategory?: string;
+    tag?: string;
+    oldTag?: string;
+    billable?: string;
+    oldBillable?: string;
+};
 
-let currentUserEmail;
-let currentUserAccountID;
-let isAnonymousUser;
+type Participant = {
+    accountID: number;
+    alternateText: string;
+    firstName: string;
+    icons: Avatar[];
+    keyForList: string;
+    lastName: string;
+    login: string;
+    phoneNumber: string;
+    searchText: string;
+    selected: boolean;
+    text: string;
+};
+
+function isTypeTransaction(arg: Transaction | Record<string, never>): arg is Transaction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+function isTypeReportAction(arg: ReportAction | Record<string, never>): arg is ReportAction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+let currentUserEmail: string | undefined;
+let currentUserAccountID: number | undefined;
+let isAnonymousUser = false;
 
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
+    callback: (value) => {
         // When signed out, val is undefined
-        if (!val) {
+        if (!value) {
             return;
         }
 
-        currentUserEmail = val.email;
-        currentUserAccountID = val.accountID;
-        isAnonymousUser = val.authTokenType === 'anonymousAccount';
+        currentUserEmail = value.email;
+        currentUserAccountID = value.accountID;
+        isAnonymousUser = value.authTokenType === 'anonymousAccount';
     },
 });
 
-let allPersonalDetails;
-let currentUserPersonalDetails;
+let allPersonalDetails: OnyxCollection<PersonalDetails>;
+let currentUserPersonalDetails: OnyxEntry<PersonalDetails | undefined>;
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-    callback: (val) => {
-        currentUserPersonalDetails = lodashGet(val, currentUserAccountID, {});
-        allPersonalDetails = val || {};
+    callback: (value) => {
+        currentUserPersonalDetails = value?.[currentUserAccountID ?? ''];
+        allPersonalDetails = value ?? {};
     },
 });
 
-let allReports;
+let allReports: OnyxCollection<Report>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
     callback: (val) => (allReports = val),
 });
 
-let doesDomainHaveApprovedAccountant;
+let doesDomainHaveApprovedAccountant = false;
 Onyx.connect({
     key: ONYXKEYS.ACCOUNT,
-    waitForCollectionCallback: true,
-    callback: (val) => (doesDomainHaveApprovedAccountant = lodashGet(val, 'doesDomainHaveApprovedAccountant', false)),
+    // waitForCollectionCallback: true,
```

Comment:
@kubabutkiewicz Are you sure it needs to be commented?

Reply:
this was just for test and didn't uncomment it , but I don't know how to solve it because it produce a TS error , because the only valid value for this is `false` as `false` is only valid for collections and `ONYXKEYS.ACCOUNT` is not a collection.